plugins/woocommerce/src/Blocks/BlockPatterns.php

```diff
@@ -243,12 +243,21 @@ public function register_ptk_patterns() {
 	 * @return array The parsed patterns.
 	 */
 	private function parse_categories( array $patterns ) {
```

Comment:
Even though we check before its usage that `$patterns` is an array, do you think we should add more type checking in here prior to processing?

```php
if ( ! is_array( $patterns ) || empty( $patterns ) ) {
    return array();
}
```

Reply:
I personally think its good practice even if we have type checked the value before passing it to the method to also type check within the method just incase some of this code gets refactored in future and this trips us up again.